packages/design-system/src/components/HelpDrawer/HelpDrawer.jsx

```diff
@@ -24,6 +25,53 @@ export class HelpDrawer extends React.PureComponent {
 
   componentDidMount() {
     if (this.headingRef) this.headingRef.focus();
+
+    const eventAction = 'onComponentDidMount';
+
+    /* Send analytics event for helpdrawer open */
+    this.sendAnalyticsEvent(eventAction);
+  }
+
+  componentWillUnmount() {
+    const eventAction = 'onComponentWillUnmount';
+
+    /* Send analytics event for helpdrawer close */
+    this.sendAnalyticsEvent(eventAction);
+  }
+
+  sendAnalyticsEvent(eventAction) {
+    const eventProps = this.props.analytics && this.props.analytics[eventAction];
+    const sendAnalytics =
+      typeof eventProps !== 'boolean' || (typeof eventProps === 'boolean' && eventProps !== false);
+
+    if (window.utag && sendAnalytics) {
+      sendTealiumEvent(this.getAnalyticsPayload(eventAction));
+    }
+  }
+
+  getAnalyticsPayload(eventAction) {
```

Comment:
I dont think there needs to be another function here. Since you already defined a `sendAnalyticsEvent` helper function, i think it's fine to include this code there. This is a more concise example:
 
```
const analyticsOverrides= get(this.props.analytics, eventAction);
const analyticsDisabled = analyticsObject === false;

if (window.utag && !analyticsDisabled) {
    const payload = merge(this.getDefaultAnalytics(eventAction), analyticsOverrides));
    sendTealiumEvent(payload);
}
```